lib/rails_cursor_pagination/timestamp_cursor.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module RailsCursorPagination
+  # Cursor class that's used to uniquely identify a record and serialize and
+  # deserialize this cursor so that it can be used for pagination.
+  # This class expects the `order_field` of the record to be a timestamp and is
+  # to be used only when sorting a
+  class TimestampCursor < Cursor
+    class << self
+      # Decode the provided encoded cursor. Returns an instance of this
+      # +RailsCursorPagination::Cursor+ class containing both the ID and the
+      # ordering field value. The ordering field is expected to be a timestamp
+      # and is always decoded in the UTC timezone.
+      #
+      # @param encoded_string [String]
+      #   The encoded cursor
+      # @param order_field [Symbol]
+      #   The column that is being ordered on. It needs to be a timestamp of a
+      #   class that responds to `#strftime`.
+      def decode(encoded_string:, order_field:)
+        decoded = JSON.parse(Base64.strict_decode64(encoded_string))
+
+        new(
+          id: decoded[1],
+          order_field: order_field,
+          order_field_value: Time.at(decoded[0].to_r / (10**6)).utc
```

Comment:
We should maybe add a comment here explaining why this uses `#to_r` instead of just having a float (precision)..